apps/fz_http/lib/fz_http/api_tokens/api_token.ex

```diff
@@ -0,0 +1,51 @@
+defmodule FzHttp.ApiTokens.ApiToken do
+  @moduledoc """
+  Stores API Token metadata to check for revocation.
+  """
+  use FzHttp, :schema
+  import Ecto.Changeset
+
+  @max_per_user 10
+
+  schema "api_tokens" do
+    field :expires_at, :utc_datetime_usec
+
+    # User-friendly way to set expires_at
+    field :expires_in, :integer, virtual: true, default: 30
+
+    belongs_to :user, FzHttp.Users.User
+
+    timestamps(updated_at: false)
+  end
+
+  def changeset(api_token, attrs, opts \\ []) do
+    api_token
+    |> cast(attrs, ~w[
+      user_id
+      expires_in
+      expires_at
+    ]a)
+    |> validate_required([:user_id, :expires_in])
+    |> validate_number(:expires_in, greater_than_or_equal_to: 1, less_than_or_equal_to: 90)
+    |> resolve_expires_at()
+    |> validate_required(:expires_at)
+    |> assoc_constraint(:user)
+    |> maybe_validate_count_per_user(@max_per_user, opts[:count_per_user])
```

Comment:
A race is still possible here because counts are returned from a separate transaction, so you can concurrently create more than 10 of them. If we want to keep the counts and make sure the limit is reliable, fixing it won't be easy - we need to fetch the count in the same DB transaction and lock something during the creation (as a semaphore).

Reply:
Good point. I can note that this is an imperfect solution for us to tackle later -- `max_device_count` violates the same principles. If we're off by one or two records it's ok, not ideal, but would that suffice for now? If someone hits this issue in the wild maybe it's better to fail hard in some way so we can learn more about their use case. That would be a lot of admins and/or api tokens.